### PR TITLE
Readding CloudStack e2e tests with newer K8s versions

### PR DESCRIPTION
**Pull request by maxdrib**
*Issue #, if available:*
https://github.com/aws/eks-anywhere/issues/4281

*Description of changes:*
Re-adds tests that were removed as part of K8s 1.20 deprecation in PR https://github.com/aws/eks-anywhere/pull/4247, but without any replacements. These tests must be skipped for now as we do not currently have CloudStack VM images available which support K8s 1.22 and 1.23 (see issue https://github.com/aws/eks-anywhere/issues/4282). We also need to add some placeholder values in secrets manager for the required environment variables to allow the other tests to run.

Not including tests for K8s 1.24 as that is not yet tested with the CloudStack provider and is out of scope for this change.

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cmd/integration_test/build/buildspecs/test-eks-a-cli.yml [hunk 1]
```diff
@@ -87,6 +87,8 @@ env:
     T_CLOUDSTACK_MANAGEMENT_SERVER: "cloudstack_ci_beta_connection:management_server"
     T_CLOUDSTACK_MANAGEMENT_SERVER_2: "cloudstack_ci_beta_connection:management_server_2"
     T_CLOUDSTACK_TEMPLATE_REDHAT_1_21: "cloudstack_ci_beta_connection:template_redhat_1_21"
+    T_CLOUDSTACK_TEMPLATE_REDHAT_1_22: "cloudstack_ci_beta_connection:template_redhat_1_22"
+    T_CLOUDSTACK_TEMPLATE_REDHAT_1_23: "cloudstack_ci_beta_connection:template_redhat_1_23"
```
[the remark this answers is not in the document]
Comment (maxdrib) [reply cmd/integration_test/build/buildspecs/test-eks-a-cli.yml:91]: Just now, yes 👍

File: test/e2e/upgrade_test.go [hunk 2]
```diff
@@ -398,6 +398,75 @@ func TestVSphereKubernetes123BottlerocketTo124StackedEtcdUpgrade(t *testing.T) {
...
+	)
+}
+
+func TestCloudStackKubernetesUnstacked122RedhatTo123Upgrade(t *testing.T) {
+	provider := framework.NewCloudStack(t, framework.WithCloudStackRedhat122())
+	test := framework.NewClusterE2ETest(
+		t,
+		provider,
+		framework.WithClusterFiller(api.WithExternalEtcdTopology(1)),
+		framework.WithClusterFiller(api.WithKubernetesVersion(v1alpha1.Kube120)),
```
Comment (mitalipaygude) [inline test/e2e/upgrade_test.go:425]: This should be `Kube122` I believe.

**Pull request merged**